unity-environment/Assets/ML-Agents/Scripts/Academy.cs

```diff
@@ -216,24 +239,75 @@ void InitializeEnvironment()
         // Retrieve Brain and initialize Academy
         List<Brain> brains = GetBrains(gameObject);
         InitializeAcademy();
-
-        // Check for existence of communicator
-        communicator = new ExternalCommunicator(this);
-        if (!communicator.CommunicatorHandShake())
+        MLAgents.Communicator.Communicator communicator= null;
+        try
+        {
+            communicator = new MLAgents.Communicator.RpcCommunicator(
+                new MLAgents.Communicator.CommunicatorParameters
+                {
+                Port = ReadArgs()
+                });
+        }
+        catch
         {
             communicator = null;
+            foreach (Brain b in brains)
+            {
+                if (b.brainType == BrainType.External)
+                {
+                    communicator = new MLAgents.Communicator.RpcCommunicator(
+                        new MLAgents.Communicator.CommunicatorParameters
+                        {
+                            Port = 5005
+                        });
+                }
+            }
+
         }
 
+        brainBatcher = new MLAgents.Batcher(communicator);
+
         // Initialize Brains and communicator (if present)
         foreach (Brain brain in brains)
         {
-            brain.InitializeBrain(this, communicator);
+            brain.InitializeBrain(this, brainBatcher);
         }
         if (communicator != null)
         {
             isCommunicatorOn = true;
-            communicator.InitializeCommunicator();
-            communicator.UpdateCommand();
+
+            MLAgents.Communicator.AcademyParameters academyParameters=
+                new MLAgents.Communicator.AcademyParameters();
+            academyParameters.Name = gameObject.name;
+            academyParameters.Version = "API-3";
```

Comment:
Is this constant used anywhere else? Creating a single global const will save you version-hell.